### DISCUSSION TITLE
Casadi .map don't keep options

### DISCUSSION THREAD

**Discussion by charleskhazoom** [casadi/casadi#3808] [Q&A] [answered]
I am unable to keep the options passed to a .casadi function after calling .map. I tried to pass options when calling map like so  :
`Fserial = f.map('Fserial','serial',n_batch,[],[], opts);` This runs without error, but the options are not used properly.

I also tried `Fserial.change_option('record_time', true);`, but this option cannot be changed.

Is there an alternative. The goal is to use the `record_time` option to properly benchmark parallelization of casadi functions from Matlab.

**Comment by tmmsartor** [comment, marked as answer]
To my knowledge is not possible to specify option when defining a function map, what you can do it wrap the input-output obtained by `.map()` in a new function with arbitrary options. You can do it manually but there exists also a convenient method to do so, namely `.wrap_as_needed()` accepting an options dictionary as parameter.
To illustrate better, here is a short example, this is in Python but should work also in Matlab after translating it
```python
import numpy as np
import time
from casadi import *

N = 100

x = SX.sym('x')
y = x
X = MX.sym('X',N)
for k in range(100000): y = sin(y)
f0 = Function('f', [x], [y], {"record_time":True})
print()

# serial
t0 = time.time()
for i in range(N): res = f0(1.0)
t1 = time.time()
print(f"f0 total elapsed {(t1-t0):.5f} seconds")
print(f"f0 average elapsed {(t1-t0)/N:.6f} seconds")
print(f"f0 stats: {f0.stats()}")
print()

# parallel evaluation
fMap = f0.map(N,"openmp")
dummyInput = np.linspace(0.0, 2.0*np.pi, N)
outMap = fMap(dummyInput)
print("fMap stats: ", fMap.stats())

# parallel evaluation with options
opts = {"record_time": True}
fMapWrap = fMap.wrap_as_needed(opts)
outMap = fMapWrap(dummyInput)
print("fMapWrap stats: ", fMapWrap.stats())
```
